core/api-server/tests/exec-raw.js

```diff
@@ -272,6 +272,29 @@ describe('Executions', () => {
             const response = await request(options);
             expect(response.body).to.have.property('jobId');
         });
+        it('should insert gateway', async () => {
```

Comment:
you should add tests for 
- gateway without a name
- gateway not exists
- gateway already exists